internal/web/middleware.go

```diff
@@ -124,24 +128,38 @@ func AuthMW(log *logrus.Entry, tm token.Manager) Middleware {
 				parsedToken, err := tm.ParseWithClaims(tkn, JWTSigningSecret, &claims)
 				if err != nil {
 					w.WriteHeader(http.StatusUnauthorized)
-					fwd := forwardedHeader(r)
-					pluginID := normalizePluginID(fwd["by"])
+					fwd := ForwardedHeader(r)
+					pluginID := NormalizePluginID(fwd["by"])
+
+					// an empty plugin ID indicates an admin token
+					if pluginID == "" {
```

Comment:
In this case when there is an error with the admin token, we should still write an error back to karavictl with `JSONErrorResponse`.

As is, the token could be bad but karavictl wouldn't get an error back.

Reply:
fixed